OpenRA.Mods.Common/Traits/Carryall.cs

```diff
@@ -164,6 +164,18 @@ public virtual bool AttachCarryable(Actor self, Actor carryable)
 
 			CarryableOffset = OffsetForCarryable(self, carryable);
 			aircraft.LandAltitude = aircraft.LandAltitude - new WDist(CarryableOffset.Z);
+
+			aircraft.GetPosition = () =>
```

Comment:
Our usual strategy for cases like this is to define a new interface (maybe `IAircraftCenterOffset` and `IPreventAircraftLanding`?) and have `Aircraft` ask the trait(s) for their values when needed.

This helps decouple the traits, improves moddability, and for things like offsets makes it easy to aggregate over multiple active traits.

Reply:
Done. Although it should be noted that we don't just prevent landing in some terrain, we also need to enable landing in other terrain (amphibious vehicles for instance).